### PR TITLE
Fix endian dependent color reading

### PR DESCRIPTION
**Pull request by Eathox**
## Issue

When reading the HEX colors from a xpm42 file `strtol` is used over the whole HEX sequence, `strtol` treats this whole sequence as one number. Therefor when the endian doesn't match the presumed RGBA order the result is not the expected color.

Sample image with incorrect endian:
incorect

## Solution

Instead of using `strtol` over the whole HEX sequence, use it only over the individual color channels and then manually bit shift the channels to the expected location. 

Sample image with HEX channel correction:
corect

### CODE REVIEW

File: src/textures/mlx_xpm42.c [hunk 1]
```diff
@@ -37,6 +37,17 @@
  * straight forward to this format however.
  */
 
+/**
+ * Parses HEX color channel e.g: "0F"
+ * 
+ * @param channel The 2 character string to parse.
+ * @return Int value of the channel.
+ */
+static uint8_t parse_hex_channel(char *channel) {
```
Comment (W2Wizard) [inline src/textures/mlx_xpm42.c:46]: Absolute heresy!

File: src/textures/mlx_xpm42.c [hunk 2]
```diff
@@ -37,6 +37,17 @@
...
+	char temp_chan[] = {channel[0], channel[1], '\0'};
+	return (uint8_t)strtol(temp_chan, NULL, 16);
```
Comment (W2Wizard) [inline src/textures/mlx_xpm42.c:48]: I know its technically not necessary but in the Wiki I describe the repo code style, I prefer to keep the return values in brackets: `return (...);`

Comment (Eathox) [reply src/textures/mlx_xpm42.c:48]: Ah! totally forgot to read the wiki, maybe you should consider mentioning the wiki code style in [CONTRIBUTING.md](https://github.com/W2Codam/MLX42/blob/master/CONTRIBUTING.md)?

File: src/textures/mlx_xpm42.c (hunk 1 above)
Comment (W2Wizard) [reply src/textures/mlx_xpm42.c:46]: Also remove the `Int` in the return value, I think its redundant.

File: src/textures/mlx_xpm42.c [hunk 3]
```diff
@@ -37,6 +37,17 @@
...
+}
+
 /**
  * Parses the XPM coolor value entry e.g: ".X #00FF00FF"
```
Comment (W2Wizard) [inline src/textures/mlx_xpm42.c:52]: Just noticed this smoll typo, feel free to fix this, or not.

File: src/textures/mlx_xpm42.c (hunk 2 above)
Comment (W2Wizard) [reply src/textures/mlx_xpm42.c:48]: Good point! Will do.

File: src/textures/mlx_xpm42.c [hunk 4]
```diff
@@ -57,8 +68,14 @@ static bool mlx_insert_xpm_entry(xpm_t* xpm, char* line, uint32_t* ctable, size_
 	if (!isspace(line[xpm->cpp]) || line[xpm->cpp + 1] != '#' || !isalnum(line[xpm->cpp + 2]))
 		return (false);
 
+	uint32_t color = 0;
+	size_t start_offset = xpm->cpp + 2;
+	color |= parse_hex_channel(line + start_offset) << 24;
+	color |= parse_hex_channel(line + start_offset + 2) << 16;
+	color |= parse_hex_channel(line + start_offset + 4) << 8;
+	color |= parse_hex_channel(line + start_offset + 6);
```
Comment (W2Wizard) [inline src/textures/mlx_xpm42.c:76]: Prefix with `mlx_`

File: src/textures/mlx_xpm42.c (hunk 3 above)
Comment (Eathox) [reply src/textures/mlx_xpm42.c]: Addressed in: 446ef18

File: src/textures/mlx_xpm42.c (hunk 4 above)
Comment (Eathox) [reply src/textures/mlx_xpm42.c]: Addressed in: 4e97dde

File: src/textures/mlx_xpm42.c (hunk 2 above)
Comment (Eathox) [reply src/textures/mlx_xpm42.c]: Fixed in: f5f359e

File: src/textures/mlx_xpm42.c (hunk 1 above)
Comment (Eathox) [reply src/textures/mlx_xpm42.c]: I believe I addressed this in: f5f359e, unless you meant that you want me to move this function to _mlx_utils.c_

Comment (W2Wizard) [reply src/textures/mlx_xpm42.c]: Just the `*` symbol must be with the type :D

Comment (Eathox) [reply src/textures/mlx_xpm42.c]: Old habits die hard: 87879ce

**Comment by W2Wizard** [comment]
Looks all good! Thank you 👍

**Pull request merged**